WordPress/Classes/Models/AbstractPost.m

```diff
@@ -452,8 +452,17 @@ - (BOOL)isDraft
 
 - (void)publishImmediately
 {
-    self.dateModified = [NSDate date];
-    [self setDateCreated:self.dateModified];
+    // If the post was modified later than it was created
+    // then this is a post that was published at some point
+    // we will not update the dateCreated
```

Comment:
A bit nitpicky perhaps, but if there is a case where calling `publishImmedately` will not set the date to `now` then maybe the method should be also be renamed for clarity?